acceptance/scripts/generic/testrun.sh

```diff
@@ -14,9 +16,58 @@ export BEAKER_HELPER="acceptance/lib/helper.rb"
 
 bundle install --path vendor/bundle
 
-bundle exec genconfig2 $GENCONFIG_LAYOUT > $BEAKER_CONFIG
+#TODO: Someday, when our Rakefile is refactored, this section will go away
+BEAKER="bundle exec beaker --debug"
+BEAKER="$BEAKER --type aio"
+BEAKER="$BEAKER --keyfile $BEAKER_KEYFILE"
+BEAKER="$BEAKER --helper $BEAKER_HELPER"
+BEAKER="$BEAKER --options-file $BEAKER_OPTIONS"
+BEAKER="$BEAKER --post-suite $BEAKER_POSTSUITE"
+BEAKER="$BEAKER --load-path acceptance/lib"
+
+
+case $1 in
+  -p | --p* )
+  bundle exec genconfig2 $GENCONFIG_LAYOUT > $BEAKER_CONFIG
+    
+  if [ -z "$PACKAGE_BUILD_VERSION" ]; 
+    #TODO: curl builds.puppetlabs.lan/puppetserver.  Parse HTML, find most recent folder name.
+    then echo "$0: PACKAGE_BUILD_VERSION not set.  Please export PACKAGE_BUILD_VERSION."
+    exit -1;
+  fi
+  
+  BEAKER="$BEAKER --pre-suite $BEAKER_PRESUITE"
+  BEAKER="$BEAKER --config $BEAKER_CONFIG"
+  BEAKER="$BEAKER --preserve-hosts"
+  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
+  $BEAKER
+  echo "Beaker exited with $?"
+  cp log/latest/hosts_preserved.yml .
+  ;;
+
 
-bundle exec beaker \
+-r | --r* )
+  if [ ! -s ./hosts_preserved.yml ];
+  then echo "$0: Can not find hosts_preserved.yml; can not run without presuite./n \
+    Either put a hosts_preserved.yml or use this script with -p to create new hosts and run the pre-suite against them"
+    exit -1;
+  fi
+
+  if [ -z $BEAKER_TESTSUITE ];
+    then echo "$0 BEAKER_TESTSUITE not set.  Export BEAKER_TESTSUITE."
+    exit -1;
+  fi
+  
+  BEAKER="$BEAKER --config hosts_preserved.yml"
+  BEAKER="$BEAKER --preserve-hosts"
+  BEAKER="$BEAKER --tests $BEAKER_TESTSUITE"
+  $BEAKER
+  ;;
+
+
+* )
+  # run it the old way  
+  bundle exec beaker \
```

Comment:
Might be nice to have this build off of `$BEAKER` like the other cases do - to cut down on the repetition a bit.